### PR TITLE
Update github action

### PR DESCRIPTION
**Pull request by sheplu**
- Update actions/stale from v4 to v8

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
@sheplu did you read through all the breaking changes? I didn't double check them

**Comment by sheplu** [comment]
I did read the list of breaking changes and it seems none will be breaking / changing how the action work for us https://github.com/actions/stale/blob/main/CHANGELOG.md

But there is also some new options that may or may not be useful for the repo

**Review by VoltrexKeyva — APPROVED** [review]

File: .github/workflows/close-stale-issues.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
     if: github.repository == 'nodejs/help'
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/stale@v4
+      - uses: actions/stale@v8
```
Comment (bmuenzenmeyer) [inline .github/workflows/close-stale-issues.yml:25]: OpenSSF and GitHub best practices would suggest we use a direct sha

```suggestion
      - uses: actions/stale@1160a2240286f5da8ec72b1c0816ce2481aabf84 # v8.0.0
```

**Review by bmuenzenmeyer — APPROVED** [review]

File: .github/workflows/close-stale-issues.yml (hunk 1 above)
Comment (ljharb) [reply .github/workflows/close-stale-issues.yml:25]: True, but those “best practices” basically kill free-flowing bugfixes and security updates. I usually recommend only pinning to a sha when not doing so risks leakage of secrets.

Comment (sheplu) [reply .github/workflows/close-stale-issues.yml:25]: are we good to keep it like that @bmuenzenmeyer ?

**Comment by maanu1234** [comment]
Great

**Review by maanu1234 — COMMENTED** [review]
Tku

**Review by Hinaroos67 — APPROVED** [review]

**Comment by Hinaroos67** [comment]
> - Update actions/stale from v4 to v8

**Pull request merged**